File: epi_judge_cpp/element_appearing_once.cc
#include <vector>

#include "test_framework/generic_test.h"
using namespace std;

int FindElementAppearsOnce(const vector<int> &nums) {
    unordered_map<int, int> num_to_count_map;
    for (int num : nums) {
        num_to_count_map[num]++;
    }
    for (auto &[num, count] : num_to_count_map) {
        if (count == 1) return num;
    }
    return 0;
}

int main(int argc, char *argv[]) {
    std::vector<std::string> args{argv + 1, argv + argc};
    std::vector<std::string> param_names{"A"};
    return GenericTestMain(args, "element_appearing_once.cc",
                           "element_appearing_once.tsv", &FindElementAppearsOnce,
                           DefaultComparator{}, param_names);
}

File: epi_judge_java/epi/ElementAppearingOnce.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class ElementAppearingOnce {

    @EpiTest(testDataFile = "element_appearing_once.tsv")
    public static int findElementAppearsOnce(List<Integer> nums) {
        Map<Integer, Integer> numToCountMap = new HashMap<>();
        for (int num : nums) {
            numToCountMap.merge(num, 1, Integer::sum);
        }
        for (var entry : numToCountMap.entrySet()) {
            if (entry.getValue() == 1) return entry.getKey();
        }
        return -1;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "ElementAppearingOnce.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}
